x-pack/plugin/mapper-exponential-histogram/src/main/java/org/elasticsearch/xpack/exponentialhistogram/ExponentialHistogramFieldMapper.java

```diff
@@ -259,6 +265,7 @@ public void parse(DocumentParserContext context) throws IOException {
                 return;
             }
 
+            Double sum = null;
```

Comment:
Maybe change to something like this to avoid the allocation. OTOH, parsing this is probably already pretty allocation intensive so it might not matter.
```
double sum = 0;
boolean hasSum = false;
``